.github/workflows/build-and-test.yaml

```diff
@@ -16,7 +16,7 @@ on:
 
 env:
   RUST_FMT: nightly-2023-04-01-x86_64-unknown-linux-gnu
-  RUST_VERSION: "1.66"
+  RUST_VERSION: "1.69"
```

Comment:
The date-time thing requires 1.67 only.

Also this version should be updated in Cargo.toml.